### PR TITLE
feat(CFW): add new data_source firewalls and new resource protection_rule

### PR DESCRIPTION
**Pull request by Jason-Zhang9309**
**What this PR does / why we need it**:

add new data_source firewalls

**Which issue this PR fixes**:
*(optional, in `fixes #<issue number>(, fixes #<issue_number>, ...)` format, will close that issue when PR gets merged)*
fixes #xxx

**Special notes for your reviewer**:

**Release note**:

```release-note
1. add new data_source firewalls;
2. and new resource protection_rule;
```

## PR Checklist

* [x] Tests added/passed.
* [x] Documentation updated.
* [x] Schema updated.

## Acceptance Steps Performed

```
make testacc TEST='./huaweicloud/services/acceptance/cfw' TESTARGS='-run TestAccDatasourceFirewalls_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cfw -v -run TestAccDatasourceFirewalls_basic -timeout 360m -parallel 4
=== RUN   TestAccDatasourceFirewalls_basic
=== PAUSE TestAccDatasourceFirewalls_basic
=== CONT  TestAccDatasourceFirewalls_basic
--- PASS: TestAccDatasourceFirewalls_basic (12.56s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cfw       12.670s

make testacc TEST='./huaweicloud/services/acceptance/cfw' TESTARGS='-run TestAccProtectionRule_basic'
==> Checking that code complies with gofmt requirements...
TF_ACC=1 go test ./huaweicloud/services/acceptance/cfw -v -run TestAccProtectionRule_basic -timeout 360m -parallel 4
=== RUN   TestAccProtectionRule_basic
=== PAUSE TestAccProtectionRule_basic
=== CONT  TestAccProtectionRule_basic
--- PASS: TestAccProtectionRule_basic (26.49s)
PASS
ok      github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance/cfw       26.652s
```

### CODE REVIEW

File: docs/data-sources/cfw_firewalls.md [hunk 1]
```diff
@@ -0,0 +1,127 @@
...
+## Argument Reference
+
+The following arguments are supported:
+
+* `region` - (Optional, String) Specifies the region in which to query the data source.
+  If omitted, the provider-level region will be used.
+
+* `service_type` - (Required, Int) Specifies the service type. The value can be:
+  **0**: North-south firewall;
+  **1**: East-west firewall;
```
Comment (ShiChangkuo) [inline docs/data-sources/cfw_firewalls.md]:
```suggestion
  + **1**: East-west firewall;
```

File: docs/data-sources/cfw_firewalls.md [hunk 2]
```diff
@@ -0,0 +1,127 @@
...
+
+<a name="firewalls_GetFirewallInstanceResponseRecordProtectObjectVO"></a>
+The `GetFirewallInstanceResponseRecordProtectObjectVO` block supports:
+
+* `object_id` - The protected object ID.
+
+* `object_name` - The protected object name.
+
+* `type` - The project type. The options are as follows:
+  **0**: north-south;
```
Comment (ShiChangkuo) [inline docs/data-sources/cfw_firewalls.md]: ditto

File: docs/data-sources/cfw_firewalls.md [hunk 3]
```diff
@@ -0,0 +1,127 @@
...
+* `eip_count` - The number of EIPs.
+
+* `log_storage` - The log storage.
+
+* `version` - The firewall version. The value can be 0 (standard edition), 1 (professional edition),
+  2 (platinum edition), or 3 (basic edition).
+
+* `vpc_count` - The number of VPCs.
+
+<a name="firewalls_GetFirewallInstanceResponseRecordProtectObjectVO"></a>
```
Comment (ShiChangkuo) [inline docs/data-sources/cfw_firewalls.md]: **VO** is unnecessary?

File: huaweicloud/services/acceptance/cfw/data_source_huaweicloud_cfw_firewalls_test.go [hunk 4]
```diff
@@ -0,0 +1,38 @@
...
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+)
+
+func TestAccDatasourceFirewalls_basic(t *testing.T) {
+	rName := "data.huaweicloud_cfw_firewalls.test"
+	dc := acceptance.InitDataSourceCheck(rName)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheck(t) },
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/cfw/data_source_huaweicloud_cfw_firewalls_test.go]: it's better to add a preCheck if the CFW instance was not created by terraform

File: docs/data-sources/cfw_firewalls.md (hunk 1 above)
Comment (Jason-Zhang9309) [reply docs/data-sources/cfw_firewalls.md]: fixed

File: docs/data-sources/cfw_firewalls.md (hunk 2 above)
Comment (Jason-Zhang9309) [reply docs/data-sources/cfw_firewalls.md]: fixed

File: docs/data-sources/cfw_firewalls.md (hunk 3 above)
Comment (Jason-Zhang9309) [reply docs/data-sources/cfw_firewalls.md]: fixed

File: huaweicloud/services/acceptance/cfw/data_source_huaweicloud_cfw_firewalls_test.go (hunk 4 above)
Comment (Jason-Zhang9309) [reply huaweicloud/services/acceptance/cfw/data_source_huaweicloud_cfw_firewalls_test.go]: fixed

File: docs/resources/cfw_protection_rule.md [hunk 5]
```diff
@@ -0,0 +1,189 @@
...
+The [Rule Source Address](#ProtectionRule_RuleSourceAddress) structure is documented below.
+
+* `destination` - (Required, String) The destination configuration.
+The [Rule Destination Address](#ProtectionRule_RuleDestinationAddress) structure is documented below.
+
+* `status` - (Required, Int) The rule status.
+  0: disabled; 1: enabled.
+
+* `long_connect_enable` - (Required, Int) Whether to support persistent connections.
+  0: supported; 1: not supported.
```
Comment (Lance52259) [inline docs/resources/cfw_protection_rule.md]: Maybe the list style is clearer.
```
+ **0**: disabled.
+ **1**: enabled. 
```

File: docs/resources/cfw_protection_rule.md [hunk 6]
```diff
@@ -0,0 +1,189 @@
...
+* `long_connect_time_hour` - (Optional, Int) The persistent connection duration (hour).
+
+* `long_connect_time_minute` - (Optional, Int) The persistent connection duration (minute).
+
+* `long_connect_time_second` - (Optional, Int) The persistent Connection Duration (second).
+
+* `description` - (Optional, String) The description.
+
+* `direction` - (Optional, Int) The direction.
+  0 inbound,1 outbound
```
Comment (Lance52259) [inline docs/resources/cfw_protection_rule.md]: ditto. see `top` attribute

File: docs/resources/cfw_protection_rule.md [hunk 7]
```diff
@@ -0,0 +1,189 @@
...
+
+## Attributes Reference
+
+In addition to all arguments above, the following attributes are exported:
+
+* `id` - The resource ID.
+
+## Import
+
+The protectionrule can be imported using `object_id`, `id`, separated by slashes, e.g.
```
Comment (Lance52259) [inline docs/resources/cfw_protection_rule.md]:
```suggestion
The protectionrule can be imported using `object_id`, `id`, separated by a slash, e.g.
```
just one slash: `<object_id>/<id>`

File: docs/resources/cfw_protection_rule.md (hunk 5 above)
Comment (Jason-Zhang9309) [reply docs/resources/cfw_protection_rule.md]: fixed

File: docs/resources/cfw_protection_rule.md (hunk 6 above)
Comment (Jason-Zhang9309) [reply docs/resources/cfw_protection_rule.md]: fixed

File: docs/resources/cfw_protection_rule.md (hunk 7 above)
Comment (Jason-Zhang9309) [reply docs/resources/cfw_protection_rule.md]: fixed

File: huaweicloud/services/acceptance/cfw/data_source_huaweicloud_cfw_firewalls_test.go [hunk 8]
```diff
@@ -0,0 +1,38 @@
...
+
+	"github.com/huaweicloud/terraform-provider-huaweicloud/huaweicloud/services/acceptance"
+)
+
+func TestAccDatasourceFirewalls_basic(t *testing.T) {
+	rName := "data.huaweicloud_cfw_firewalls.test"
+	dc := acceptance.InitDataSourceCheck(rName)
+
+	resource.ParallelTest(t, resource.TestCase{
+		PreCheck:          func() { acceptance.TestAccPreCheckCfw(t) },
```
Comment (ShiChangkuo) [inline huaweicloud/services/acceptance/cfw/data_source_huaweicloud_cfw_firewalls_test.go]: acceptance.TestAccPreCheck(t) is missing

Comment (Jason-Zhang9309) [reply huaweicloud/services/acceptance/cfw/data_source_huaweicloud_cfw_firewalls_test.go]: fixed

**Pull request merged**